src/sidebar/components/NotebookView.js

```diff
@@ -45,7 +46,21 @@ function NotebookView({ loadAnnotationsService, streamer }) {
   // likely to be focused (eg. because the notebook has been configured to
   // display a particular group when launched), we can optimistically fetch
   // annotations from that group.
-  const groupId = focusedGroup?.id || store.directLinkedGroupId();
+  const [groupId, setGroupId] = useState(
+    focusedGroup?.id || store.directLinkedGroupId()
+  );
```

Comment:
Since you are setting the focused group below in response to a fragment change, and this component is re-rendered whenever the focused group changes, you don't need to `groupId` to be a separate piece of state here.